test/pages/products/index.spec.tsx

```diff
@@ -1,21 +1,26 @@
 import { ROW_NUMBERS } from '@mocks/hooks';
 import Products from '@pages/products/index';
-import { render, screen } from '@test/utils';
+import { render, waitFor } from '@test/utils';
 
 jest.mock('@lib/hooks', () => require('@mocks/hooks'));
 
 describe('Product List', () => {
     test('renders correctly', async () => {
-        const { container } = render(<Products />);
+        const { container, getByRole } = render(<Products />);
         // Wait for table to be rendered
-        await screen.findByRole('table');
-
-        const rowsLength = screen.getAllByRole('row').length - 1; // Rows - 1 (table header)
+        await waitFor(() => getByRole('table'));
```

Comment:
We shouldn't be using `waitFor`. What we had before, using `screen` was better.

Reply:
to be honest, I don't think it matters. `findBy`includes  `waitFor `. React testing library even includes this pattern in their docs (https://testing-library.com/docs/react-testing-library/example-intro#act). Outside of semantic preference, I don't believe there's any kind of performance hit one way or the other. The only real reason I can think of to use `screen` is to not list the queries as part of the render result, but that comes at the expense conciseness/ destructuring. 

The impetus behind this change was that `findByRole` was causing a `TypeError: Cannot read property 'createEvent' of null` error so I split it into `waitFor` and `getByRole`.

Reply:
Kent C. Dodds writes about this https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-waitfor-to-wait-for-elements-that-can-be-queried-with-find. The main two points is that it's quicker to read (easier to debug) and you'll get a better error message using `findBy*` queries.

I believe there is an underlying issue for the `TypeError` I pulled your PR down and switched it over and wasn't receiving such error. (I was using node@16 + npm@8 though). We should dig into why you are getting that error.